stock_picking_product_interchangeable/models/stock_move.py

```diff
@@ -0,0 +1,75 @@
+from odoo import models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def _prepare_interchangeable_products(self, mode, for_qty):
+        """
+        Preparing products for current product replacement
+        :param move_id: stock.move record
+        :param mode: stock.picking.type substitute_products_mode value
+        :param for_qty: product qty for replacement
+        :return: list of tuple [(product_obj, count), ...]
+        """
+        qty = for_qty
+        if qty >= 0:
+            return False, False
+        products = []
+        interchangeable_products_qty = 0
+        for product in self.product_id.product_interchangeable_ids:
+            available_qty = product.immediately_usable_qty
+            if available_qty > 0 > qty:
+                product_qty = abs(qty) if available_qty + qty >= 0 else available_qty
+                products.append((product, product_qty))
+                qty += product_qty
+                interchangeable_products_qty += product_qty
```

Comment:
I think you can move the calculation of `interchangeable_products_qty` from this method

Reply:
Resolved